src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -712,13 +712,27 @@ object FunctionRegistry {
     flattenOrNull[Set, Any](Set.newBuilder[Any], s)
   )(setHr(setHr(TTHr)), setHr(TTHr))
 
+  registerMethod("keys", (m: Map[Any, Any]) =>
+    m.keysIterator.toArray[Any]: IndexedSeq[Any]
+  )(mapHr(TTHr, TUHr), arrayHr(TTHr))
+
+  registerMethod("keySet", (m: Map[Any, Any]) =>
+    m.keySet
+  )(mapHr(TTHr, TUHr), setHr(TTHr))
+
+  registerMethod("get", (m: Map[Any, Any], key: Any) =>
+    m.get(key).orNull
+  )(mapHr(TTHr, TUHr), TTHr, TUHr)
+
   registerMethod("mkString", (a: IndexedSeq[String], d: String) => a.mkString(d))(
     arrayHr(stringHr), stringHr, stringHr)
   registerMethod("mkString", (s: Set[String], d: String) => s.mkString(d))(
     setHr(stringHr), stringHr, stringHr)
 
   registerMethod("contains", (s: Set[Any], x: Any) => s.contains(x))(setHr(TTHr), TTHr, boolHr)
-  registerMethod("contains", (d: Map[String, Any], x: String) => d.contains(x))(dictHr(TTHr), stringHr, boolHr)
+  registerMethod("contains", (d: Map[Any, Any], x: Any) => d.contains(x))(mapHr(TTHr, TUHr), TTHr, boolHr)
+
+  registerMethod("keys", (d: Map[Any, Any]) => d.keySet)(mapHr(TTHr, TUHr), setHr(TTHr))
```

Comment:
Seems like it's already defined above with a different return type / implementation. As mentioned above, I like this one though :)